frameworks/AutoGluon/setup.sh

```diff
@@ -9,16 +9,17 @@ fi
 
 # creating local venv
 . ${HERE}/../shared/setup.sh ${HERE} true
-#if [[ -x "$(command -v apt-get)" ]]; then
-#    SUDO apt-get install -y libomp-dev
+
+# Below fixes seg fault on MacOS due to bug in libomp: https://github.com/awslabs/autogluon/issues/1442
 if [[ -x "$(command -v brew)" ]]; then
-    brew install libomp
+    wget https://raw.githubusercontent.com/Homebrew/homebrew-core/fb8323f2b170bd4ae97e1bac9bf3e2983af3fdb0/Formula/libomp.rb
+    brew install libomp.rb
+    rm libomp.rb
 fi
 
 PIP install --upgrade pip
 PIP install --upgrade setuptools wheel
-PIP install "mxnet<2.0.0"
-PIP install "scikit-learn-intelex<2021.3"
+PIP install "scikit-learn-intelex<2021.6"
```

Comment:
Why are these dependencies installed separately?

Reply:
Currently it isn't part of our build-from-source install by default, and I haven't tested across windows, mac, and ARM to ensure this isn't unstable in those configs. It is stable on Linux though. If it is important, I could switch to having it in the same line as the general pip install (`pip install autogluon.tabular[all,skex]`), but would prefer it to be as it is now for stability reasons outside of the benchmark.

Reply:
Basically the alternative is to shift it to be specified in two locations:

for pypi install:

instead of:
`PIP install autogluon`

it would be:

`PIP install autogluon.tabular[all,skex]`

and for source install:

instead of

`PIP install -e tabular/[all]`

it would be:

`PIP install -e tabular/[all,skex]`

Reply:
I was just trying to consider ways to make the relationship of the installation script to current versions less hard-coded. In principle, we would like to be able to provide backward compatibility to older versions of the frameworks when reasonable.

Reply:
This is hard currently because of the existence of `PIP` being different from `pip`. I'm unsure the reasoning for this in AutoMLBenchmark, but it is the reason I can't use AutoGluon's default build-from-source logic mentioned in our install instructions: https://auto.gluon.ai/stable/index.html

This is the file I'd like to call to build AG from source: https://github.com/awslabs/autogluon/blob/master/full_install.sh

However, I can't because this file uses `pip` while I need to use `PIP` in AutoMLBenchmark.

If this can be resolved then it should be do-able to have backwards compatibility

Reply:
@Innixma see my suggestion below.

> I'm unsure the reasoning for this in AutoMLBenchmark

it was mainly to hide the automatic creation of the venv, but still be able to access both default python exec and the venv python exec if needed.

the exec is stored in `$py_exec` variable if you need it, for example to set a custom PATH as I suggest below. I think this should be enough for your needs.

Reply:
Thanks for this! I wasn't so savvy with bash so this was very helpful.

Reply:
Re scikit-intelex, I've resolved this by not hardcoding the version but instead using the version defined via the `skex` extra_dependencies key.